cassandra/src/main/java/org/apache/calcite/adapter/cassandra/CassandraRules.java

```diff
@@ -258,7 +258,8 @@ private static boolean isEqualityOnKey(RexNode node, List<String> fieldNames,
   public static class CassandraProjectRule extends CassandraConverterRule {
     /** Default configuration. */
     private static final Config DEFAULT_CONFIG = Config.INSTANCE
-        .withConversion(LogicalProject.class, Convention.NONE,
+        .withConversion(LogicalProject.class, p -> p.getCorrelVariable() == null
+                || p.getVariablesSet().isEmpty(), Convention.NONE,
```

Comment:
Do we need to check both `p.getCorrelVariable()` and `p.getVariablesSet()`? From the API and existing implementations I get the impression that `p.getVariablesSet()` is a superset of `p.getCorrelVariable()`.

Reply:
This looks like attempt to check if collection is null or empty, but something went wrong... Fixed this to verify both `p.getCorrelVariable()` and `p.getVariablesSet()`